### PR TITLE
Feat: like vsc-leetcode-cli use @lc code=start and @lc code=end to pick the code 

### PR DESCRIPTION
**Pull request by wendajiang**
1. edit add file include extra information
2. remove test file, and use memory all_cases
3. use scraper crate parser the desc html fragment

Besides, as I do not like auto create test file in the workspace, so this PR use Draft. And if someone need the @lc code=start feature, we can discuss how to merge to master.

PS. @lc code=start almost use for me to auto-completion code like this 
```
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int, int> mapping;

        for (int i = 0; i < nums.size(); i++) {
            mapping[nums[i]] = i;
        }
        for (int i = 0; i < nums.size(); i++) {
            int searched = target - nums[i];
            if (mapping.find(searched) != mapping.end() &&
                i != mapping[searched]) {
                result.push_back(i);
                result.push_back(mapping[searched]);
                break;
            }
        }
        return result;
    }
};
// @lc code=end
```
so that, when exec problem, only upload the content between `@lc code = start` and `@lc code=end `

### CODE REVIEW

**Comment by clearloop** [comment]
thx!! @wendajiang 

> Besides, as I do not like auto create test file in the workspace, so this PR use Draft. And if someone need the @lc code=start feature, we can discuss how to merge to master.

Understand this concern, due to some users like to review their past code, I think we can add a flag to keep it? how do you think like this ?

File: src/cmds/edit.rs [hunk 1]
```diff
@@ -22,6 +22,9 @@ use clap::{Command as ClapCommand, ArgMatches, Arg};
 /// ```
 pub struct EditCommand;
 
+pub const CODE_START: &str = r#"// @lc code=start"#;
+pub const CODE_END: &str = r#"// @lc code=end"#;
```
Comment (clearloop) [inline src/cmds/edit.rs:26]: shoud configure these in the config, the `//` can not be recognized by every programming languages

Comment (wendajiang) [reply src/cmds/edit.rs:26]: Yeah, you are right, `//` maybe wrong!

File: src/helper.rs [hunk 2]
```diff
@@ -92,167 +92,27 @@ mod filter {
...
-            3 => "³".to_string(),
-            4 => "⁴".to_string(),
-            5 => "⁵".to_string(),
-            6 => "⁶".to_string(),
-            7 => "⁷".to_string(),
-            8 => "⁸".to_string(),
-            9 => "⁹".to_string(),
-            _ => n.to_string(),
-        }
-    }
```
Comment (clearloop) [inline src/helper.rs:129]: plz check if scrapper can support this

Comment (wendajiang) [reply src/helper.rs:129]: however, I do not think is nessesary, and this relpace use `^` and `_`, if you think this must be use the utf-8 character, I can do some manual work to recover this. :)

Comment (clearloop) [reply src/helper.rs:129]: these optimizations are from https://github.com/clearloop/leetcode-cli/pull/12 and https://github.com/clearloop/leetcode-cli/pull/39, we better to keep these for respecting the previous contributors, btw, I think there is a escraper in our crate also lol https://github.com/clearloop/leetcode-cli/pull/11/files

**Pull request merged**